### PR TITLE
Request schema changes to agent task protection handlers

### PR DESCRIPTION
**Pull request by amogh09**
### Summary
1. Change the request schema of agent task protection handlers in accordance with design changes made to the ECS task protection APIs.
2. Improve logging of requests in agent task protection handlers. 
3. Make agent task protection handlers return HTTP 400 BadRequest error if taskARN is not found for the request. This indicates an invalid v3 Endpoint ID in the request.

### Testing

1. Changes have been made to unit tests. 
2. Built the agent from updated source and deployed it to an EC2 instance. Confirmed that the handlers work with new request schema by sending test requests using curl.

New tests cover the changes: yes

### Description for the changelog
Update agent task protection handlers in accordance with design changes.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go [hunk 1]
```diff
@@ -59,53 +60,56 @@ func PutTaskProtectionHandler(state dockerstate.TaskEngineState,
...
+		logger.Error("Failed to find task ARN for task protection request", logger.Fields{
+			loggerfield.Error: err,
+		})
+		return nil, http.StatusBadRequest, errors.New("Invalid request: no task was found")
 	}
 
 	task, found := state.TaskByArn(taskARN)
 	if !found {
-		return nil, fmt.Errorf("could not find task from task ARN '%v'", taskARN)
+		logger.Error("No task was found for taskARN for task protection request", logger.Fields{
```
Comment (Realmonia) [inline agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go:106]: Do you think this should be a critical? I feel like this is a big problem which essentially means our task engine state is corrupted.

Comment (amogh09) [reply agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go:106]: Yes if this happens then it's definitely a bug in our state management. I will change the log level to critical.

Comment (amogh09) [reply agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go]: Done.

**Review by Realmonia — APPROVED** [review]

File: agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go [hunk 2]
```diff
@@ -59,53 +60,56 @@ func PutTaskProtectionHandler(state dockerstate.TaskEngineState,
...
-		logger.Info("Would have called ECS.PutTaskProtection with fields", logger.Fields{
-			"cluster":                  cluster,
-			"serviceName":              task.ServiceName,
-			"taskId":                   task.Arn,
-			"protectionType":           taskProtection.GetProtectionType(),
-			"protectionTimeoutMinutes": taskProtection.GetProtectionTimeoutMinutes(),
+		logger.Info("PutTaskProtection endpoint was called", logger.Fields{
+			loggerfield.Cluster:        cluster,
+			loggerfield.TaskARN:        task.Arn,
+			loggerfield.TaskProtection: taskProtection,
```
Comment (YashdalfTheGray) [inline agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go:88]: Ooh I like this.

**Review by YashdalfTheGray — APPROVED** [review]

**Pull request merged**